packages/nextjs/src/client/index.ts

```diff
@@ -38,7 +38,12 @@ const globalWithInjectedValues = global as typeof global & {
 export function init(options: BrowserOptions): void {
   applyTunnelRouteOption(options);
   buildMetadata(options, ['nextjs', 'react']);
-  options.environment = options.environment || process.env.NODE_ENV;
+
+  options.environment =
+    options.environment ||
+    (process.env.NEXT_PUBLIC_VERCEL_ENV ? `vercel-${process.env.NEXT_PUBLIC_VERCEL_ENV}` : undefined) ||
```

Comment:
l: Can we maybe put this into a small util (even in the same file) like:

```js
function getVercelEnv(): string | undefined {
  const vercelEnv = process.env.NEXT_PUBLIC_VERCEL_ENV;
  return vercelEnv && `vercel-${vercelEnv}`;
}
```

Feels better to me IMHO than to sprinkle this through the rest of the code - then we can just do e.g.

```js
options.environment = options.environment || getVercelEnv() || process.env.NODE_ENV;
```

Reply:
It would just deduplicate this once because we have `VERCEL_ENV` for node and edge and `NEXT_PUBLIC_VERCEL_ENV` for the client.

Reply:
can't we just dynamically chose via a boolean? 

I would also like to see this as a func because this is totally something that where we only update two, but forgot to update one (because of merge conflicts, quick fix PR etc.)

Reply:
Democracy wins :p dfc5d20f744a48ca2a9833263a1f87b6fd505613